SocietalConstructionTool/SctRunner.cs

```diff
@@ -87,10 +87,14 @@ public static (string? outputText, IEnumerable<CompilerError> errors) CompileSct
             return (outputText, []);
         }
 
-        private static List<CompilerError> RunFirstPassChecks(ParserRuleContext startNode, CTableBuilder cTableBuilder)
+        private static List<CompilerError> RunFirstPassChecks(ParserRuleContext startNode, CTableBuilder cTableBuilder, SctParser parser)
```

Comment:
This can be refactored to only take the `SctParser`, and getting the `ParserRuleContext` from the `parser.start()`.